src/fftvis/core/utils.py

```diff
@@ -128,60 +132,76 @@ def get_task_chunks(nprocesses: int, nfreqs: int, ntimes: int) -> tuple[int, lis
         The number of frequency channels required.
     ntimes
         The number of time integrations required.
-        
+
     Returns
     -------
     nprocesses : int
-        The number of processes to actually use (typically the same as the input, 
+        The number of processes to actually use (typically the same as the input,
         but can be returned as 1 if sub-process threading is better)
     freq_chunks : list of slices
         A length-nprocesses list of lists of integers, where each sublist defines
         a set of frequency channels to compute.
     time_chunks : list of slices
         A length-nprocesses list of lists of integers, where each sublist defines
         a set of time integrations to compute.
+    nf : int
+        The number of frequency channels per chunk.
+    nt : int
+        The number of time integrations per chunk.
     """
     ntasks = ntimes * nfreqs
 
-    if ntasks < 2*nprocesses:
+    if ntasks < 2 * nprocesses:
         # Prefer lower-latency parallelization of the component calculations
         return 1, [slice(None)], [slice(None)], nfreqs, ntimes
-        
-    
+
     if ntimes >= nprocesses:
-        freq_chunks = [list(range(nfreqs))]*nprocesses
-    
+        freq_chunks = [list(range(nfreqs))] * nprocesses
+
     nt = int(np.ceil(ntimes / nprocesses))
     nf = nfreqs
     nfc = 1
-    size = nf*nt
+    size = nf * nt
     sizes = [size]
-    
-    while nf > 1 and (nprocesses*size) > ntasks:
+
+    while nf > 1 and (nprocesses * size) > ntasks:
         nfc += 1
         nf = int(np.ceil(nfreqs / nfc))
-        nt = int(np.ceil(ntimes / (nprocesses/nfc)))
-        size = nf*nt
+        nt = int(np.ceil(ntimes / (nprocesses / nfc)))
+        size = nf * nt
         sizes.append(size)
-        
+
     idx = np.argmin(sizes)
     nfc = 1 + idx
     nf = int(np.ceil(nfreqs / nfc))
-    nt = int(np.ceil(ntimes / (nprocesses/nfc)))
-    
-    ntc = int(np.ceil(nprocesses/nfc))
-    freq_chunks = [slice(nf*i, min(nfreqs, (i + 1)*nf)) for i in range(nfc)]*ntc
-    time_chunks = sum(([slice(i*nt, min(ntimes, (i + 1)*nt))]*nfc for i in range(ntc)), start=[])
+    nt = int(np.ceil(ntimes / (nprocesses / nfc)))
+
+    ntc = int(np.ceil(nprocesses / nfc))
+    freq_chunks = [slice(nf * i, min(nfreqs, (i + 1) * nf)) for i in range(nfc)] * ntc
+    time_chunks = sum(
+        ([slice(i * nt, min(ntimes, (i + 1) * nt))] * nfc for i in range(ntc)), start=[]
+    )
     return nprocesses, freq_chunks, time_chunks, nf, nt
 
-@nb.jit(nopython=True)
-def inplace_rot(rot: np.ndarray, b: np.ndarray):  # pragma: no cover
-    """In-place rotation of coordinates."""
+
+def inplace_rot_base(rot, b):
+    """
+    Base implementation of in-place rotation of coordinates.
```

Comment:
There will need to be an empty line between the first summary and the longer summary.